profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PlainTextAllocationEventExporter.java

```diff
@@ -0,0 +1,137 @@
+import com.splunk.opentelemetry.profiler.util.StackSerializer;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.sdk.logs.LogProcessor;
+import io.opentelemetry.sdk.logs.data.LogDataBuilder;
+import io.opentelemetry.sdk.resources.Resource;
+import java.time.Instant;
+import jdk.jfr.consumer.RecordedEvent;
+import jdk.jfr.consumer.RecordedStackTrace;
+import jdk.jfr.consumer.RecordedThread;
+
+public class PlainTextAllocationEventExporter implements AllocationEventExporter {
+  private static final AttributeKey<Long> ALLOCATION_SIZE_KEY =
+      AttributeKey.longKey("memory.allocated");
+
+  private final StackSerializer stackSerializer;
+  private final LogProcessor logProcessor;
+  private final LogDataCommonAttributes commonAttributes;
+  private final Resource resource;
+
+  private PlainTextAllocationEventExporter(Builder builder) {
+    this.stackSerializer = builder.stackSerializer;
+    this.logProcessor = builder.logProcessor;
+    this.commonAttributes = builder.commonAttributes;
+    this.resource = builder.resource;
+  }
+
+  @Override
+  public void export(RecordedEvent event, AllocationEventSampler sampler, SpanContext spanContext) {
+    RecordedStackTrace stackTrace = event.getStackTrace();
+    if (stackTrace == null) {
+      return;
+    }
+    Instant time = event.getStartTime();
+    String body = buildBody(event, stackTrace);
+
+    AttributesBuilder builder =
+        commonAttributes
+            .builder(event.getEventType().getName())
+            .put(DATA_TYPE, ProfilingDataType.ALLOCATION.value())
+            .put(DATA_FORMAT, DataFormat.TEXT.value())
+            .put(ALLOCATION_SIZE_KEY, event.getLong("allocationSize"));
+    if (sampler != null) {
+      sampler.addAttributes((k, v) -> builder.put(k, v), (k, v) -> builder.put(k, v));
+    }
+    Attributes attributes = builder.build();
+
+    LogDataBuilder logDataBuilder =
+        LogDataBuilder.create(resource, INSTRUMENTATION_LIBRARY_INFO)
+            .setEpoch(time)
+            .setBody(body)
+            .setAttributes(attributes);
+
+    if (spanContext != null && spanContext.isValid()) {
+      logDataBuilder.setSpanContext(spanContext);
+    }
+
+    logProcessor.emit(logDataBuilder.build());
+  }
+
+  private String buildBody(RecordedEvent event, RecordedStackTrace stackTrace) {
+    String stack = stackSerializer.serialize(stackTrace);
+    RecordedThread thread = event.getThread();
+    String name = thread == null ? "unknown" : thread.getJavaName();
+    long id = thread == null ? 0 : thread.getJavaThreadId();
+    String result = "\"" + name + "\"" + " #" + id;
+    if (thread != null) {
+      result += " nid=0x" + Long.toHexString(thread.getOSThreadId());
+    }
+    result += "\n";
+    result += "   java.lang.Thread.State: UNKNOWN\n" + stack;
+    return result;
```

Comment:
I would leave it as it is because `StringBuilder` reduces the readability. Also the plan is to use `pprof` exporter by default so this class really won't be used by anything else besides our tests if all goes as planned. Perhaps we can even delete it at some point.